handlers/oauth/database.go

```diff
@@ -0,0 +1,52 @@
+package oauth
+
+import (
+	"errors"
+
+	"github.com/speedrun-website/leaderboard-backend/model"
+	"gorm.io/gorm"
+)
+
+type gormContainer struct {
+	DB *gorm.DB
+}
+
+type OauthStore interface {
+	GetUserByTwitterID(string) (*model.User, error)
+	CreateUser(model.User) (*model.User, error)
+}
+
+var Oauth OauthStore
+
+//initGormContainer initializes the oauth store with a gorm container
+func InitGormContainer(db *gorm.DB) {
+	Oauth = &gormContainer{
+		DB: db,
+	}
+}
+
+//GetUserByTwitterID fetches a user by their twitter ID
+func (s gormContainer) GetUserByTwitterID(twitterID string) (*model.User, error) {
+	var maybeExistingUser model.User
+	result := s.DB.Model(&model.User{}).Where("twitter_id = ?", twitterID).First(&maybeExistingUser)
+
+	// No error means we found a user
+	if result.Error == nil {
+		return &maybeExistingUser, nil
+	}
+	isNotFoundError := errors.Is(result.Error, gorm.ErrRecordNotFound)
```

Comment:
nit: As a style thing I personally prefer just doing `errors.Is` inline in the `if` condition instead of the transient variable. I think it reads slightly better. Not a requirement though if you feel differently.